### DISCUSSION TITLE
Interrompere script

### DISCUSSION THREAD

**Discussion by Matty1184** [indomus/forum#3147] [HUB: Home Assistant]
Ciao a tutti, sto facendo alcune prove con alcuni script... ma non riesco a venirne a capo!
Ho creato uno script che apre tutte le tapparelle delle finestre, una dopo l'altra al raggiungimento della posizione 100%aperta, una catena praticamente.
Mi è capitato che a volte il 100% venga manualmente interrotto in modo involontario da noi tramite la pressione del pulsante di stop manuale della tapparella, il problema è che lo script resta in esecuzione e a parte non completare il ciclo (che non interessa) se dopo 1 ora la tapparella raggiunge il 100% riparte ovviamente il resto dello script.
Vorrei che dopo X minuti se lo script non è terminato si disabiliti e abbandona il ciclo.
Ho provato con Delay ma resta comunque appeso.
Grazie a tutti in anticipo

**Comment by snakuzzo** [comment]
Posta lo YAML dello script che hai creato (e usa il markdown)

**Reply by Matty1184** [reply]
Ciao, 
ecco lo script 
```
alias: Apertura Totale Tapparelle
sequence:
  - device_id: 408002c49b95cb38f77fe4b133894e1b
    domain: cover
    entity_id: cover.tapparella_bagno_terra_window_covering
    type: set_position
    position: 100
  - wait_template: >-
      {{
      is_state_attr('cover.tapparella_bagno_terra_window_covering','current_position',
      100) }}
  - device_id: e2f7a80c3315bd28c9a1592fa075325b
    domain: cover
    entity_id: cover.tapparella_cucina_window_covering
    type: set_position
    position: 100
  - wait_template: >-
      {{
      is_state_attr('cover.tapparella_cucina_window_covering','current_position',
      100) }}
    continue_on_timeout: false
  - device_id: 61311b944b313afcf07f8c1f7789e55a
    domain: cover
    entity_id: cover.tapparella_vetrata_soggiorno_window_covering
    type: set_position
    position: 100
  - wait_template: >-
      {{
      is_state_attr('cover.tapparella_vetrata_soggiorno_window_covering','current_position',
      100) }}
    continue_on_timeout: false
  - device_id: 5b5234919ea84f1a71d33776b75d1225
   
.... HO TONCATO APPOSTITAMENTE IL CODICE IN QUANTO TUTTO UGUALE ...

    position: 100
  - wait_template: >-
      {{
      is_state_attr('cover.tapparella_scale_window_covering','current_position',
      100) }}
    continue_on_timeout: false
  - service: notify.notify
    data:
      message: Tapparelle tutte Aperte
mode: single
icon: mdi:window-shutter-open
```

**Reply by snakuzzo** [reply]
Innanzi tutto, ti consiglierei di scrivere le automazioni in YAML basandoti sulle entità, evitando l'uso dei device. 

Tornando al tuo problema...Così come l'hai impostato, se interrompi l'apertura di una tapparella, l'automazione resterà in attesa all'infinito, o comunque finché la tapparella che stava aprendo non viene aperta al 100%.
Dovresti impostare per ogni wait template un timeout...di seguito il link alla doc ufficiale...

https://www.home-assistant.io/docs/scripts/#wait-timeout
